### DISCUSSION TITLE
pnpm install: Ignoring not compatible lockfile

### DISCUSSION THREAD

**Discussion by MrSaints** [vercel/community#4408] [Help]
### Summary

Hi all,

When deploying locally with just `vercel`, I noticed the following in the logs:

```
Running "vercel build"
00:08:20.818 | Vercel CLI 32.4.1
00:08:21.190 | Running "install" command: `pnpm install`...
00:08:21.718 | WARN  Ignoring not compatible lockfile at /vercel/path0/pnpm-lock.yaml
```

pnpm version: 8.8.0
Node version: v18.18.0
lockfileVersion: '6.0'

Based on https://vercel.com/docs/deployments/builds/package-managers, it should be supported, so I'm not sure how to resolve this warning. Any ideas?

### Steps to Reproduce

1. Set-up Next.js w/ App Router (versions mentioned above in summary)
2. Set `pnpm install` as the install command, and `pnpm build` for the build
3. Run `vercel`
4. Check logs

**Comment by lucas-0liveira** [comment]
hello, this moment It seems like there's a warning message appearing when deploying locally with Vercel. The warning is related to the use of pnpm as the package manager and it mentions that the lockfile is being ignored.

According to the Vercel documentation, pnpm should be supported, so it's a bit puzzling why this warning is showing up. I'm not entirely sure about the exact solution to this, but I can suggest a few steps you might try to resolve it:

Check pnpm Version: Ensure that you're using a compatible version of pnpm. It looks like you're using pnpm version 8.8.0, which should be fine, but it might be worth double-checking if there's a newer version available.

Verify Lockfile: Make sure that the pnpm-lock.yaml file is correctly set up and compatible with the project. You might want to review its contents and structure.

Vercel CLI and Configuration: Ensure that you're using the correct version of the Vercel CLI (32.4.1 in this case). Additionally, review your Vercel project configuration to see if there are any specific settings related to package managers that need attention.

Community Forums or Support: Since this seems like a specific technical issue, you might want to try searching or asking on the Vercel community forums or reaching out to their support for more specific guidance.

Fallback to npm or yarn (if possible): If you're not making extensive use of pnpm-specific features, consider temporarily switching to npm or yarn to see if that resolves the warning.

Remember to always back up your project or use version control before making significant changes to your build process. This way, you can easily revert if something goes wrong. Good luck with resolving this issue!

**Reply by MrSaints** [reply]
Appreciate your ChatGPT response, but I wouldn't be posting here if I hadn't already tried some of those suggestions.

**Comment by amyegan** [comment]
I'm not sure what you have already tried. If you let us know what you already did, or share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), we might be able to come up with better suggestions 🙏  

The first easy troubleshooting step with a lockfile issue is usually to rebuild it. If you generate a new lockfile, do you still get this error?
